src/deployment/flavors.md

```diff
@@ -3,41 +3,226 @@ title: Creating flavors for Flutter
+
+![Step 2 using flavors image.](/assets/images/docs/flavors/step2-using-flavors.png){:width="100%"}  
+
+</li>
+<li markdown=1>
+
+Add the display name to **info.plist**. Create a new key in **info.plist** called **Bundle Display Name** 
+with the value `$(PRODUCT_NAME)`. 
+
+![Step 3 using flavors image.](/assets/images/docs/flavors/step3-using-flavors.png){:width="100%"}    
+
+</li>
+</ol>
+
+Now you have set up your development flavor by making a `free` scheme 
+in Xcode and setting the build configurations for that scheme. 
+
+For more information, skip to the [“Launching your app flavors”][] 
+section at the end of this document.
+
+### Plugin configurations
+
+If your app has a Flutter plugin, you will need to update the `ios/Podfile`. 
+
+1. In `ios/Podfile` change the default of **Debug**, **Profile**, and **Release** 
+to match the Xcode build configurations for the `free` scheme.
+
+```ruby
+project 'Runner', {
+  'Debug-free' => :debug,
+  'Debug-production' => :debug,
+  'Profile-free' => :release,
+  'Profile-production' => :release,
+  'Release-free' => :release,
+  'Release-production' => :release,
+}
+```
+
+## Using flavors in Android
+
+Setting up flavors in Android can be done in your project's 
+**build.gradle** file.
+
+1. In your IDE, inside your Flutter project, 
+navigate to **android** > **app** > **build.gradle**.   
+
+![Step 1 using flavors in Android image.](/assets/images/docs/flavors/android-step1.png){:width="100%"}    
+
+Add a flavor object with the specified environments along with values for 
+**dimension**, **resValue**, and **applicationId**. 
+The name of the application for each build is located in **resValue** 
+(as shown in the screenshot below from VSCode).
+## Setting up launch configurations
+
+Next, you'll add a **launch.json** file allows you to run the command 
+`flutter run --flavor [environment name]`.    
+In VSCode, set up the launch configurations as follows:    
+1. In the root directory of your project, add a folder called **.vscode**.    
+2. Inside the **.vscode** folder, create a file named **launch.json**.    
+3. In the **launch.json** file, add a configuration object for each flavor. 
+Each configuration has a **name**, **request**, **type**, **program**, 
+and **args** key.
+
+You can now run the terminal command 
+`flutter run --flavor [environment name]` or you can set up a run 
+configuration in your IDE.
+
+```ruby
+productFlavors {
+    development {
+        dimension "default"
+        resValue "string", "app_name", "dev free flavor example"
+        applicationId ".dev"
+    }
+    production {
+        dimension "default"
+        resValue "string", "app_name", "prod free flavor example"
+        applicationId ""
+    }
+}
+``` 
+
+## Launching your app flavors
+
+1. Once the flavors are set up, modify the Dart code in 
+**lib** > **main.dart** to consume the flavors. 
+2. Test the setup using `flutter run --flavor [environment name]` 
```

Comment:
I don't think "environment name" is very clear, can we use the actual flavor name we just made?
```suggestion
2. Test the setup using `flutter run --flavor free` 
```